Firmware/ultralcd.cpp

```diff
@@ -1962,26 +2014,49 @@ extern char* __malloc_heap_start;
 extern char* __malloc_heap_end;
 #endif //DEBUG_STACK_MONITOR
 
+//! @brief Show Debug Information
+//!
+//! @code{.unparsed}
+//! |01234567890123456789|
+//! |RAM statistics      |	c=20 r=1
+//! | SP_min:        0000|	c=14 r=1
+//! | heap_start:    0000|	c=14 r=1
+//! | heap_end:      0000|	c=14 r=1
+//! ----------------------
+//! @endcode
+//! @todo Positioning of the messages and values on LCD aren't fixed to their exact place. This causes issues with translations.
 static void lcd_menu_debug()
 {
 #ifdef DEBUG_STACK_MONITOR
 	lcd_home();
-	lcd_printf_P(PSTR("RAM statistics\n" " SP_min: 0x%04x\n" " heap_start: 0x%04x\n" " heap_end: 0x%04x"), SP_min, __malloc_heap_start, __malloc_heap_end);
+	lcd_printf_P(PSTR("RAM statistics\n" " SP_min: 0x%04x\n" " heap_start: 0x%04x\n" " heap_end: 0x%04x"), SP_min, __malloc_heap_start, __malloc_heap_end);  ////c=20 r=1  ////c=14 r=1  ////c=14 r=1  ////c=14 r=1
 #endif //DEBUG_STACK_MONITOR
 
 	menu_back_if_clicked_fb();
 }
 #endif /* DEBUG_BUILD */
 
+//! @brief Show Temperatures
+//!
+//! @code{.unparsed}
+//! |01234567890123456789|
+//! | Nozzle:        000D|	c=14 r=1
+//! | Bed:           000D|	c=14 r=1
+//! | Ambient:       000D|	c=14 r=1
+//! | PINDA:         000D|	c=14 r=1
+//! ----------------------
+//! D - Degree sysmbol		LCD_STR_DEGREE
+//! @endcode
+//! @todo Positioning of the messages and values on LCD aren't fixed to their exact place. This causes issues with translations.
 static void lcd_menu_temperatures()
 {
 	lcd_timeoutToStatus.stop(); //infinite timeout
 	lcd_home();
-	lcd_printf_P(PSTR(" %S:   %d%c \n" " %S:      %d%c \n"), _i("Nozzle"), (int)current_temperature[0], '\x01', _i("Bed"), (int)current_temperature_bed, '\x01');
+	lcd_printf_P(PSTR(" %S:   %d%c \n" " %S:      %d%c \n"), _i("Nozzle"), (int)current_temperature[0], '\x01', _i("Bed"), (int)current_temperature_bed, '\x01');  ////c=14 r=1  ////c=14 r=1
```

Comment:
again, this menu is subject to improvements, the hard-coded layout breaks in every other language

Reply:
Here another example:
```
	lcd_printf_P(PSTR(ESC_H(1,0) "%S:" ESC_H(16,0) "%-3d%c" ESC_H(1,1) "%S:" ESC_H(16,1) "%-3d%c"), _i("Nozzle"), (int)current_temperature[0], '\x01', _i("Bed"), (int)current_temperature_bed, '\x01');
	lcd_printf_P(PSTR(ESC_H(1,2) "%S:" ESC_H(16,2) "%-3d%c" ESC_H(1,3) "%S:" ESC_H(16,3) "%-3d%c"), _i("Ambient"), (int)current_temperature_ambient, '\x01', _i("PINDA"), (int)current_temperature_pinda, '\x01');
	lcd_printf_P(PSTR(ESC_H(1,2) "%S:" ESC_H(16,2) "%-3d%c"), _i("PINDA"), (int)current_temperature_pinda, '\x01');
```

Reply:
Since we removed the escape sequences for the 3.8 (which saved 1KB along with other @leptun 's changes) this won't work anymore. Positioning the cursor absolutely may be a way around the formatting too, it just won't be so clear ;)

Reply:
@DRracer @3d-gussner 
VT100 codes: I was thinking that we could make for a easy and straightforward way to pad with spaces by using something like lcd_padding(int col, int row). This should add spaces until the target position is reached on the line. This is just a proposal for now and it is not set in stone yet. I'll try implementing these menus in a easy way soon and see what I get.

Reply:
@leptun slow down ;) ... no need for elaborate solutions in this case - printf can be made to justify a string left and pad it with spaces by using %-12.12S ... only the ':' must be a part of the justified string. I believe, we can make the whole formatting using just one printf call, which saves a lot of FLASH.